src/main/java/io/stargate/sgv2/jsonapi/api/model/command/deserializers/FilterClauseDeserializer.java

```diff
@@ -93,7 +175,16 @@ private ComparisonExpression createComparisonExpression(Map.Entry<String, JsonNo
 
   private static Object jsonNodeValue(String path, JsonNode node) {
     if (path.equals(DocumentConstants.Fields.DOC_ID)) {
-      return DocumentId.fromJson(node);
+      if (node.getNodeType() == JsonNodeType.ARRAY) {
```

Comment:
Comment here could be useful to indicate it's needed for `$in` operator, as otherwise `_id` cannot be an Array value (and it's not here, it's being compared)?

Reply:
Done